ddl/ddl.go

```diff
@@ -465,13 +465,19 @@ func (dc *ddlCtx) removeJobCtx(job *model.Job) {
 	delete(dc.jobCtx.jobCtxMap, job.ID)
 }
 
-func (dc *ddlCtx) jobContext(jobID int64) *JobContext {
+func (dc *ddlCtx) jobContext(jobID int64, reorgMeta *model.DDLReorgMeta) *JobContext {
 	dc.jobCtx.RLock()
 	defer dc.jobCtx.RUnlock()
+	var ctx *JobContext
 	if jobContext, exists := dc.jobCtx.jobCtxMap[jobID]; exists {
-		return jobContext
+		ctx = jobContext
+	} else {
+		ctx = NewJobContext()
+	}
+	if reorgMeta != nil && len(ctx.resourceGroupName) == 0 {
```

Comment:
```suggestion
	if reorgMeta != nil {
```
No need to check ctx's resource-group, it may be the default value `default`

Reply:
Is `default` necessary? If it is, we need add a lock for it. Otherwise, it will occur data race.

Reply:
Is there any case that depend on the ctx.resourceGroupName here? otherwise, should always override with reorgMeta's